Utils/StatisticsCalculator.cs

```diff
@@ -0,0 +1,193 @@
+          } else {
+            Console.WriteLine("ALERT (GetHighestEmissionActivityByUserID): ActivityType read from query does not exists in emissionDict: " + activityType);
+          }
+        }
+      } else {
+        connection.Close();
+        return null;
+      }
+      connection.Close();
+    }
+
+    // Calculate percentage
+    double sum = emissionsDict.Values.Sum();
+    emissionsDict["transport"] = emissionsDict["transport"] / sum;
+    emissionsDict["food"] = emissionsDict["food"] / sum;
+    emissionsDict["utility"] = emissionsDict["utility"] / sum;
+
+    // Determine the highest emission activity
+    KeyValuePair<string, double> highestEmissionActivity = emissionsDict
+      .OrderByDescending(keyValue => keyValue.Value)
+      .FirstOrDefault();
+    return new EmissionStatistic
+    {
+      Statistic = "Highest Emission Activity",
+      Topic = char.ToUpper(highestEmissionActivity.Key[0]) + highestEmissionActivity.Key.Substring(1),
+      Stat = highestEmissionActivity.Value.ToString("0.00")
+    };
+  }
+
+  public EmissionStatistic GetAverageEmissionByUserID(int userID)
+  {
+    DateTime currentDate = DateTime.UtcNow;
+    DateTime oneWeekAgo = currentDate.AddDays(-7);
+
+    // Average the sum of total emission with how many entries retrieved
+    double average = GetAverageEmissionBetweenTwoDate(currentDate, oneWeekAgo, userID);
+    if (average == -1)
+    {
+      return null;
+    }
+
+    return new EmissionStatistic
+    {
+      Statistic = "Average Emission",
+      Topic = "Within 1 Week",
+      Stat = average.ToString("0.00")
+    };
+  }
+
+  public EmissionStatistic GetEmissionDifferenceByUserID(int userID)
+  {
+    DateTime currentDate = DateTime.UtcNow;
+    DateTime oneWeekAgo = currentDate.AddDays(-7);
+    DateTime twoWeekAgo = currentDate.AddDays(-14);
+
+    // Retrieve DailyEmission.TotalEmission with a current week
+    double currentAverageEmission = GetAverageEmissionBetweenTwoDate(currentDate, oneWeekAgo, userID);
+    if (currentAverageEmission == -1)
+    {
+      return null;
+    }
+
+    double previousWeekAverageEmission = GetAverageEmissionBetweenTwoDate(oneWeekAgo, twoWeekAgo, userID);
+    if (previousWeekAverageEmission == -1)
+    {
+      return null;
+    }
+
+    double difference = currentAverageEmission - previousWeekAverageEmission;
+    string stat = difference.ToString("0.00");
+    if (difference >= 0)
+    {
+      stat = "+" + difference.ToString("0.00");
+    }
+    return new EmissionStatistic
+    {
+      Statistic = "Emission Difference",
+      Topic = "Between 2 Weeks",
+      Stat = stat
+    };
+  }
+
+  public double GetAverageEmissionBetweenTwoDate(DateTime currentDate, DateTime previousDate, int userID)
+  {
+    double totalEmission = 0;
+    int numEntries = 0;
+    double averageEmission = -1;
+    // Retrieve DailyEmission.TotalEmission with a week
+    MySqlConnection connection = DatabaseConnecter.MySQLDatabase();
+
+    using (connection)
```

Comment:
Use Using with the creation of the connection object in line 158, saves some space